Dependency generation issue

This line: https://github.com/boutproject/BOUT-dev/blob/next/make.config.in#L306
leads to an error when compiling on the YPI machines:

```
  Compiling  nc_format.cxx
make[4]: m2c: Command not found
<builtin>: recipe for target '/opt/york/phys/pkg/Modules/netcdf/4.4.1.1/1/gcc-5.4.0/include/netcdf' failed
make[4]: *** [/opt/york/phys/pkg/Modules/netcdf/4.4.1.1/1/gcc-5.4.0/include/netcdf] Error 127
../../../make.config:115: recipe for target 'netcdf4' failed
make[3]: *** [netcdf4] Error 2
../../make.config:115: recipe for target 'impls' failed
make[2]: *** [impls] Error 2
```

This thread (https://lists.mcs.anl.gov/pipermail/petsc-users/2010-November/007299.html) indicates that the problem may be an implicit rule for building modula-2 .mod files.

Some key parts of my `make.config` file are:

```
BOUT_FLAGS              = $(CXXFLAGS) -g -O2 -O -DCHECK=2 -DSIGHANDLE -DLOGCOLOR -DBOUT_VERSION_STRING="\"4.1.2\"" -DBOUT_VERSION_DOUBLE=4.12  -DBACKTRACE -DNCDF4 -DHDF5 -DLAPACK -DBOUT_HAS_PVODE  -std=c++11

CXX                     = /opt/york/phys/pkg/Modules/openmpi/1.10.7/1/gcc-5.4.0/bin/mpicxx

-include .*.mk
```
Tested on the `sawtooth` machine with :
```
Currently Loaded Modulefiles:
  1) null             3) its-hdd          5) openmpi/1.10.7   7) szip/2.1.1       9) netcdf/4.4.1.1  11) fftw/3.3.6
  2) its-net          4) gcc/5.4.0        6) zlib/1.2.11      8) hdf5/1.8.18     10) sundials/2.7.0  12) petsc/3.7.6
```
Configuring with no arguments:
```
Configuration summary

  PETSc support: no
  SLEPc support: no
  IDA support: no
  CVODE support: no
  ARKODE support: no
  NetCDF support: yes
  Parallel-NetCDF support: no
  HDF5 support: yes (parallel: no)
  MUMPS support: no
  Lapack support: yes
  Scorep support: no
```

--- a/make.config.in
+++ b/make.config.in
@@ -288,7 +288,6 @@ endif
 %.o: $(BOUT_CONFIG_FILE) %.cxx
 	@echo "  Compiling " $(@F:.o=.cxx)
 	@$(CXX) $(BOUT_INCLUDE) $(BOUT_FLAGS) -c $(@F:.o=.cxx) -o $@
-	@$(CXX) $(BOUT_INCLUDE) $(BOUT_FLAGS) -MF.$(@F:.o=.mk) -MM -c $(@F:.o=.cxx) || exit 0
 ifeq ("$(TARGET)","libfast")
 	test "$@" = "bout++.o" || touch $(BOUT_TOP)/lib/.last.o.file
 endif

